src/Util/Logger/WorkerLogger.php

```diff
@@ -22,26 +23,38 @@ class WorkerLogger implements LoggerInterface
 	private $workerId;
 
 	/**
-	 * @var string The current function Name of the worker
+	 * @var string The current function name of the worker
 	 */
 	private $functionName;
 
-	public function __construct(LoggerInterface $logger, $functionName)
+	/**
+	 * @param LoggerInterface $logger       The logger for worker entries
+	 * @param string          $functionName The current function name of the worker
+	 * @param int             $uidLength    The length of the generated UID
+	 *
+	 * @throws InternalServerErrorException if the uid is too long
+	 */
+	public function __construct(LoggerInterface $logger, $functionName, $uidLength = 7)
 	{
 		$this->logger = $logger;
 		$this->functionName = $functionName;
-		$this->workerId = $this->generateUid(8);
+		$this->workerId = $this->generateUid($uidLength);
 	}
 
 	/**
 	 * Generates an UID
 	 *
 	 * @param $length
 	 * @return string
+	 *
+	 * @throws InternalServerErrorException If the uid is too long
 	 */
 	private function generateUid($length)
 	{
-		return bin2hex(random_bytes($length / 2));
+		if ($length > 13) {
+			throw new InternalServerErrorException('Maximum of 13 characters for UID possible, used \'' . $length . '\'');
+		}
+		return substr(uniqid('', true), 0, $length);
```

Comment:
This isn't good either. The first 8 characters of the `uniqid()` output are a simple hex conversion of the Unix timestamp, the following 5 are the hex conversion of the microseconds. If two `generateUid(8)` are called at less than second of interval, their output will be the same.

Suggested complete change:
```php
private function generateUid($length)
{
	return substr(bin2hex(random_bytes(ceil($length / 2))), 0, $length);
}

Reply:
Hm.. Sometimes I struggle with the combination of the different transformations of string/hex... Strange enough that there isn't a standard function for such goals ^^

Reply:
The best way is to simply use ```uniqid``` without any parameter. This value is different in each microsecond: http://php.net/manual/en/function.uniqid.php

Reply:
No, the problem isn't `uniqid()` or its parameters, but truncating its output to the first few characters. `random_bytes()` output can be truncated and will still be random.